src/Core/Batch/Retry.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\Batch;
+
+/**
+ * A class for retrying the failed items.
+ */
+class Retry
+{
+    use HandleFailureTrait;
+
+    /* @var BatchRunner */
+    private $runner;
+
+    /**
+     * Initialize a BatchRunner and $failureFile.
+     *
+     * @param BatchRunner $runner [optional] **Defaults to** null.
+     */
+    public function __construct(BatchRunner $runner = null)
+    {
+        if ($runner === null) {
+            $runner = new BatchRunner();
+        }
+        $this->runner = $runner;
```

Comment:
Personal preference, but if you like you could use the elvis operator here:

`$this->runner = $runner ?: new BatchRunner();`